src/twisted/conch/test/test_window.py

```diff
@@ -66,3 +73,123 @@ def test_parent(self) -> None:
         scrolled = ScrolledArea(widget)
         self.assertIs(widget.parent, scrolled._viewport)
         self.assertIs(scrolled._viewport.parent, scrolled)
+
+
+class SelectionTests(TestCase):
+    """
+    Change focused entry in L{Selection} using function keys
+    """
+
+    def setUp(self) -> None:
+        """
+        Create L{ScrolledArea} widget with 10 elements and position selection to 5th element
+        """
+
+        seq: list[bytes] = [f"{_num}".encode("ascii") for _num in range(10)]
+        self.widget = Selection(seq, None)
+        self.widget.height = 10  # type: ignore[assignment]
+        self.widget.focusedIndex = 5
+
+    def test_selection_down_arrow(self) -> None:
+        """
+        Send DOWN_ARROW to select element just below the current one
+        """
+        self.widget.keystrokeReceived(ServerProtocol.DOWN_ARROW, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 6)
+
+    def test_selection_up_arrow(self) -> None:
+        """
+        Send UP_ARROW to select element just above the current one
+        """
+        self.widget.keystrokeReceived(ServerProtocol.UP_ARROW, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 4)
+
+    def test_selection_pgdn(self) -> None:
+        """
+        Send PGDN to select element one page down (here: last element)
+        """
+        self.widget.keystrokeReceived(ServerProtocol.PGDN, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 9)
+
+    def test_selection_pgup(self) -> None:
+        """
+        Send PGUP to select element one page up (here: first element)
+        """
+        self.widget.keystrokeReceived(ServerProtocol.PGUP, None)  # type: ignore[attr-defined]
+        self.assertIs(self.widget.focusedIndex, 0)
+
+
+class TestWidget(Widget):
+    triggered: dict[str, bool] = dict()
+
+    def func_F1(self, modifier) -> None:  # type: ignore[no-untyped-def]
+        self.triggered["F1"] = True
```

Comment:
I think that is important to record each call... in the case there is a bug and the same function is called more than once.
```suggestion
        self.triggered["F1"].append(True)
```